pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/ParticipantDataFixupService.java

```diff
@@ -3,35 +3,63 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Comparator;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 
-import com.google.gson.Gson;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.lang3.StringUtils;
 import org.broadinstitute.dsm.db.DDPInstance;
 import org.broadinstitute.dsm.db.dao.ddp.participant.ParticipantDataDao;
 import org.broadinstitute.dsm.db.dto.ddp.participant.ParticipantData;
 import org.broadinstitute.dsm.exception.DSMBadRequestException;
 import org.broadinstitute.dsm.exception.DsmInternalError;
-import org.broadinstitute.dsm.model.defaultvalues.ATDefaultValues;
-import org.broadinstitute.dsm.pubsub.WorkflowStatusUpdate;
-import org.broadinstitute.dsm.service.admin.AdminOperation;
 import org.broadinstitute.dsm.service.admin.AdminOperationRecord;
+import org.broadinstitute.dsm.service.elastic.ElasticSearchService;
+import org.broadinstitute.dsm.service.participantdata.ATParticipantDataService;
+import org.broadinstitute.dsm.service.participantdata.DuplicateParticipantData;
+import org.broadinstitute.dsm.service.participantdata.ParticipantDataService;
+import org.broadinstitute.dsm.util.ParticipantUtil;
+import org.broadinstitute.dsm.util.proxy.jackson.ObjectMapperSingleton;
 
 /**
- * AdminOperation service to asynchronously fix various ParticipantData issues
+ * AdminOperation service to asynchronously fix various ParticipantData issues.
+ * </p>
+ * The operation requires a fixupType attribute. The supported fixup types are:
+ * - atcpGenomicId: remove duplicate genomic IDs and participant exit statuses
+ * - atcpLegacyPid: for the ddp_participant_id field, replace legacy PIDs with GUIDs. Report errors
+ *   if records with legacy PIDs have the same field type ID as records with GUIDs.
  */
 @Slf4j
-public class ParticipantDataFixupService implements AdminOperation {
-    private static final Gson gson = new Gson();
+public class ParticipantDataFixupService extends ParticipantAdminOperationService {
     protected List<String> validRealms = List.of("atcp");
+    protected List<String> validFixups = List.of(FixupType.AT_GENOMIC_ID.label, FixupType.AT_LEGACY_ID.label);
     private boolean forceFlag = false;
     private DDPInstance ddpInstance;
     private Map<String, List<ParticipantData>> participantDataByPtpId;
+    private FixupType fixupType;
+    private boolean dryRun;
+
+    protected enum FixupType {
+        AT_GENOMIC_ID("atcpGenomicId"),
+        AT_LEGACY_ID("atcpLegacyId");
+
+        public final String label;
+
+        private FixupType(String label) {
+            this.label = label;
+        }
+
+        public static FixupType valueOfLabel(String label) {
+            for (FixupType fixupType : values()) {
+                if (fixupType.label.equals(label)) {
+                    return fixupType;
+                }
+            }
+            return null;
```

Comment:
Maybe throw `DsmInternalError` here instead, if there's no valid use case for a fixup with no fixupType.

Reply:
Looks good to me now!